### ISSUE TITLE
Callsign management

### ISSUE THREAD

**Issue by jxckUK** [SnailyCAD/snaily-cadv4#663]
### Describe the feature

At the moment we manage callsign allocations by using a spreadsheet and although it works, we have a computer aided dispatch system that would be much nicer to manage callsigns from.

I'm not entirely sure on the best way to implement this but effectively we need:

- Callsign allocations for each division, eg FC10-FC99.
- Callsign conditional input so they cannot be a duplicated callsign of another unit
- Increment the callsign when creating a unit based on the default divisions callsign allocation (point 1) using the same incremental logic as paired callsigns

**Comment by Dev-CasperTheGhost** [comment]
I'm not exactly sure what you mean here, you want to see a table of all callsigns, or want to have better callsign validation based on more options within department/division values?

**Comment by jxckUK** [comment]
> I'm not exactly sure what you mean here, you want to see a table of all callsigns, or want to have better callsign validation based on more options within department/division values?

Both of those! 😅

**Comment by Dev-CasperTheGhost** [comment]
>  or want to have better callsign validation based on more options within department/division values?

what would these options be?

**Comment by Dev-CasperTheGhost** [comment]
if I'm understanding correctly, you're requesting:

1. Check for duplicate callsigns?
2. `{incremental}` support for regular units?

**Comment by jxckUK** [comment]
> if I'm understanding correctly, you're requesting:
> 
> 1. Check for duplicate callsigns?
> 2. `{incremental}` support for regular units?

Yes exactly, and as you said before a page where you can see current callsign assignments. 

I'm essentially trying to get rid of our spreadsheets where we track callsign allocations.

**Comment by Dev-CasperTheGhost** [comment]
> and as you said before a page where you can see current callsign assignments. 

what about the manage units page? You can view all callsigns there too. Or does it need to more detailed.

**Comment by jxckUK** [comment]
> > and as you said before a page where you can see current callsign assignments.
> 
> what about the manage units page? You can view all callsigns there too. Or does it need to more detailed?

More detailed really, in addition I'd want to give access to a larger group of people than I'd want to allocate the manage units permission to.

**Comment by Dev-CasperTheGhost** [comment]
A new tab, there has also been a new permission added `ManageUnitCallsigns`. If a user has that permission, they can only see the `Callsign management` tab. Soon:tm:

https://user-images.githubusercontent.com/53900565/166466571-647f65be-1f8c-44ca-9693-d79191fb87ea.mov

**Comment by Dev-CasperTheGhost** [comment]
Soon:tm:: Once a unit gets updated or if a unit changes their status to on-duty, it will append an `incremental` property to the unit. No need to panic if there's an empty space for a few units.
